frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -297,6 +300,7 @@ private void trackCurrentBranchChange(GitRepository repository) {
         mostRecentlyCheckedOutBranch = repositoryCurrentBranchName;
       }
     }
+    queueRepositoryUpdateAndModelRefresh();
```

Comment:
why this change?

Reply:
So that `queueRepositoryUpdateAndModelRefresh()` happens actually **after** `mostRecentlyCheckedOutBranch` is assigned. The alternative would be sth like:

```java
@AlwaysSafe GitRepositoryChangeListener listener = repository -> {
ModalityUiUtil.invokeLaterIfNeeded(NON_MODAL,
        () -> trackCurrentBranchChange(repository)); // no queueRepositoryUpdateAndModelRefresh() call inside trackCurrentBranchChange
  queueRepositoryUpdateAndModelRefresh();
}
```

...which will also probably be okay (since `queueRepositoryUpdateAndModelRefresh()` will queue the action to execute on UI thread **after** `trackCurrentBranchChange(repository)`)... but the current approach is clearer if it comes to what happens after what :async-parrot:

Reply:
oh, okay. I see now it is moved from the other place. ok then!